everestrs/everestrs_sys/everestrs_sys.cpp

```diff
@@ -30,6 +33,41 @@ JsonBlob json2blob(const json& j) {
     return JsonBlob{vec};
 }
 
+// Below are overloads to be used with std::visit and our std::variant. We force
+// a compilation error if someone changes the underlying std::variant without
+// extending/adjusting the functions below.
+
+template <typename T, typename... VARIANT_T> struct VariantMemberImpl : public std::false_type {};
+
+template <typename T, typename... VARIANT_T>
+struct VariantMemberImpl<T, std::variant<VARIANT_T...>> : public std::disjunction<std::is_same<T, VARIANT_T>...> {};
+
+/// @brief Static checker if the type T can be converted to `ConfigEntry`.
+///
+/// We use this to detect `get_config_field` overloads which receive arguments
+/// which aren't part of our `ConfigEntry` variant.
+template <typename T> struct ConfigEntryMember : public VariantMemberImpl<T, ConfigEntry> {};
```

Comment:
in line with the c++ standard, I would call the struct `is_config_entry_member` or something similar

Reply:
furthermore, if we're using `std::visit` we got already type checking, so if we don't implement the visitor for a type, it doesn't exist. Shouldn't be that enough?  Of course there might be still implicit conversions, but that might still happen here too

Reply:
The everest projects are using CamelCase and not snake_case. 
As for why its there - it's primary task it to fail the compilation if some one removes an entry from std::variant - in this case the dev is "politely" hinted that there might be some cleanup to do in the rust-glue code

Reply:
I see, so in case we remove one type from the variant, one of the variants should fail, although they will never be called then anymore.  A single `std::is_same_v<ConfigEntry, std::variant<currently_known_types...>>` might also work here